python/pyarrow/tests/test_parquet.py

```diff
@@ -997,40 +997,162 @@ def test_read_partitioned_directory(tmpdir):
 
 
 @parquet
-def test_read_partitioned_directory_filtered(tmpdir):
-    fs = LocalFileSystem.get_instance()
-    base_path = str(tmpdir)
-
-    import pyarrow.parquet as pq
-
-    foo_keys = [0, 1]
-    bar_keys = ['a', 'b', 'c']
-    partition_spec = [
-        ['foo', foo_keys],
-        ['bar', bar_keys]
-    ]
-    N = 30
-
-    df = pd.DataFrame({
-        'index': np.arange(N),
-        'foo': np.array(foo_keys, dtype='i4').repeat(15),
-        'bar': np.tile(np.tile(np.array(bar_keys, dtype=object), 5), 2),
-        'values': np.random.randn(N)
-    }, columns=['index', 'foo', 'bar', 'values'])
-
-    _generate_partition_directories(fs, base_path, partition_spec, df)
-
-    dataset = pq.ParquetDataset(
-        base_path, filesystem=fs,
-        filters=[('foo', '=', 1), ('bar', '!=', 'b')]
+class TestParquetFilter:
```

Comment:
I'm +0 on using classes for indentation but please use new-style classes that inherit from `object`